tools/unitctl/unit-client-rs/src/unitd_docker.rs

```diff
@@ -0,0 +1,403 @@
+                typ: Some(MountTypeEnum::BIND),
+                source: Some(application.clone()),
+                target: Some("/www".to_string()),
+                read_only: Some(true),
+                ..Default::default()
+            });
+
+            let _ = docker
+                .create_image(
+                    Some(CreateImageOptions {
+                        from_image: image.as_str(),
+                        ..Default::default()
+                    }),
+                    None,
+                    None,
+                )
+                .next()
+                .await
+                .unwrap()
+                .or_else(|err| {
+                    Err(UnitClientError::UnitdDockerError {
+                        message: err.to_string(),
+                    })
+                });
+
+            let resp: ContainerCreateResponse;
+            match docker
+                .create_container::<String, String>(
+                    None,
+                    Config {
+                        image: Some(image.clone()),
+                        host_config: Some(HostConfig {
+                            network_mode: Some("host".to_string()),
+                            mounts: Some(mounts),
+                            ..Default::default()
+                        }),
+                        ..Default::default()
+                    },
+                )
+                .await
+            {
+                Err(err) => {
+                    return Err(UnitClientError::UnitdDockerError {
+                        message: err.to_string(),
+                    })
+                }
+                Ok(response) => resp = response,
+            }
+
+            let mut list_container_filters = HashMap::new();
+            list_container_filters.insert("id".to_string(), vec![resp.id]);
+            match docker
+                .list_containers::<String>(Some(ListContainersOptions {
+                    all: true,
+                    limit: None,
+                    size: false,
+                    filters: list_container_filters,
+                }))
+                .await
+            {
+                Err(e) => Err(UnitClientError::UnitdDockerError { message: e.to_string() }),
+                Ok(info) => {
+                    if info.len() < 1 {
+                        return Err(UnitClientError::UnitdDockerError {
+                            message: "couldnt find new container".to_string(),
+                        });
+                    }
+                    if info[0].names.is_none() || info[0].names.clone().unwrap().len() < 1 {
+                        return Err(UnitClientError::UnitdDockerError {
+                            message: "new container has no name".to_string(),
+                        });
+                    }
+
+                    match docker
+                        .start_container(
+                            info[0].names.clone().unwrap()[0].strip_prefix("/").unwrap(),
```

Comment:
Super small nit - you may want to use a constant to refer to the file system path separator.

Reply:
Honestly, I forget if I bothered to do that. After all, it is unlikely this will run in Windows.